app/src/main/java/com/teamounce/ounce/main/BottomSheetAdapter.kt

```diff
@@ -1,52 +1,101 @@
+    private var checkedRadioButton: CompoundButton? = null
+    private var mSelectedItem = -1
+    val bottomSheetFragment = BottomSheetFragment()
 
     override fun onCreateViewHolder(
         parent: ViewGroup,
         viewType: Int
     ): BottomSheetViewHolder {
-        val binding = ItemBottomMainBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        val binding =
+            ItemBottomMainBinding.inflate(LayoutInflater.from(parent.context), parent, false)
         return BottomSheetViewHolder(binding)
     }
 
     override fun onBindViewHolder(holder: BottomSheetViewHolder, position: Int) {
-        holder.bind(settingCareCatData[position])
+        holder.bind(bottomSheetProfileData[position], position, mSelectedItem)
+        holder.binding.catSelectBtn.setOnCheckedChangeListener(checkedChangedListener)
 
-            holder.binding.catSelectBtn.isChecked = lastSelectionPosition == position
+        if (holder.binding.catSelectBtn.isChecked) {
+            checkedRadioButton = holder.binding.catSelectBtn
+        }
+    }
 
+    override fun getItemCount(): Int {
+        return bottomSheetProfileData.size
+    }
 
+    fun changeCat(position: Int) {
+        if (mContext is MainActivity) {
+            mContext.setMainViewRetrofit(bottomSheetProfileData[position].catIndex)
+            mContext.bottomSheetFragment.dismiss()
         }
+    }
 
-    inner class BottomSheetViewHolder(var binding: ItemBottomMainBinding) : RecyclerView.ViewHolder(binding.root){
+    private val checkedChangedListener =
+        CompoundButton.OnCheckedChangeListener { compoundButton, isChecked ->
+            checkedRadioButton?.apply {
+                setChecked(!isChecked)
+            }
+            checkedRadioButton = compoundButton.apply {
+                setChecked(isChecked)
+                bottomSheetFragment.fragmentManager?.executePendingTransactions()
+                bottomSheetFragment.dialog?.setOnDismissListener {
+                    it.dismiss()
+                }
+            }
 
-        fun bind(settingCareCatData: SettingCareCatData) {
-            binding.settingCareCatData = settingCareCatData
+        }
 
-            binding.catSelectBtn.setOnClickListener {
-                lastSelectionPosition = adapterPosition
-                notifyDataSetChanged()
+    inner class BottomSheetViewHolder(var binding: ItemBottomMainBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        var sharedPreferences = SharedPreferences(mContext)
+
+        fun bind(
+            bottomSheetProfileData: BottomSheetProfileData,
+            position: Int,
+            selectedPosition: Int
+        ) {
+            binding.bottomSheetProfileData = bottomSheetProfileData
+
+            if (position == sharedPreferences.getCatPositionSelected()) {
+                binding.catSelectBtn.isChecked = true
+            } else {
+                if (selectedPosition == position) {
```

Comment:
(이건 그냥 개인적인 바람, 굳이 안 고쳐줘도 됨) 판단의 로직도 하나의 함수로 분리할 수 있어
```suggestion
                fun isSelectedPosition(selectedPosition: Int, position: Int) = (selectedPosition == possition)
                if (selectedPosition == position) {
```